tests/AgDatabaseMove.Unit/BackupOrder.cs

```diff
@@ -44,51 +60,155 @@ private static List<BackupMetadata> ListBackups()
+          foundDiff = true;
+        }
+        else if(currentBackup.BackupType == BackupFileTools.BackupType.Log) {
+          Assert.True(foundFull);
+          Assert.True(currentBackup.FirstLsn >= lastBackup.LastLsn);
+          foundLog = true;
+        }
+
+        lastBackup = currentBackup;
+        backupChain.RemoveAll(b => b.LastLsn == currentBackup.LastLsn);
+      }
+    }
+
+    public static IEnumerable<object[]> PositiveTestData => new List<object[]> {
+      new object[] { GetBackupList() },
+      new object[] { GetBackupListWithStripes() },
+      new object[] { GetBackupListWithoutDiff() },
+      new object[] { GetBackupListWithoutLogs() }
+    };
+    
+    [Theory]
+    [MemberData(nameof(PositiveTestData))]
+    public void BackupChainIsCorrect(List<BackupMetadata> backupList)
     {
       var agDatabase = new Mock<IAgDatabase>();
-      agDatabase.Setup(agd => agd.RecentBackups()).Returns(_listBackups);
+      agDatabase.Setup(agd => agd.RecentBackups()).Returns(backupList);
       var backupChain = new BackupChain(agDatabase.Object);
+      VerifyListIsAValidBackupChain(backupChain.OrderedBackups.ToList());
+    }
+
 
-      var expected = _listBackups.OrderBy(bu => bu.FirstLsn);
+    public static IEnumerable<object[]> NegativeTestData => new List<object[]> {
+      new object[] { GetBackupListWithoutFull() },
+      new object[] { new List<BackupMetadata>() }
+    };
 
-      Assert.Equal<IEnumerable>(backupChain.OrderedBackups, expected);
+    [Theory]
+    [MemberData(nameof(NegativeTestData))]
+    public void CanDetectBackupChainIsWrong(List<BackupMetadata> backupList)
+    {
+      var agDatabase = new Mock<IAgDatabase>();
+      agDatabase.Setup(agd => agd.RecentBackups()).Returns(backupList);
+      Assert.Throws<BackupChainException>(() => new BackupChain(agDatabase.Object));
     }
 
     [Fact]
     public void MissingLink()
     {
-      var backups = ListBackups().Where(b => b.FirstLsn != 126000000955200001).ToList();
+      var backups = GetBackupList().Where(b => b.FirstLsn != 126000000955200001).ToList();
       var agDatabase = new Mock<IAgDatabase>();
       agDatabase.Setup(agd => agd.RecentBackups()).Returns(backups);
 
-      var chain = new BackupChain(agDatabase.Object).OrderedBackups;
-      Assert.NotEqual(chain.Last().LastLsn, ListBackups().Max(b => b.LastLsn));
+      var chain = new BackupChain(agDatabase.Object).OrderedBackups.ToList();
+      Assert.NotEqual(chain.Last().LastLsn, GetBackupList().Max(b => b.LastLsn));
+      VerifyListIsAValidBackupChain(chain);
+    }
+
+    [Fact]
+    public void DuplicateFiles()
+    {
+      var backups = GetBackupListWithStripesAndDuplicates();
+      var agDatabase = new Mock<IAgDatabase>();
+      agDatabase.Setup(agd => agd.RecentBackups()).Returns(backups);
+      var chain = new BackupChain(agDatabase.Object).OrderedBackups.ToList();
+      Assert.Equal(chain.Count, backups.Count/2);

```

Comment:
`Assert.Equal(chain.Count, backups..GroupBy(b =>b.FirstLsn).Count);`

Reply:
ah I see - I did the direct compare cause I know what the backup list generator for this test spits back out (duplicates each file)
but you are right here, should be using a more general logic for the check in case we change the generator

Reply:
(just to note, striped backups have same LSNs so will use file name in group by clause)